symfony/src/Form/MeasurementType.php

```diff
@@ -94,11 +94,11 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             } elseif ($type === 'date') {
                 unset($fieldOptions['scale']);
                 $class = Type\DateType::class;
+                $minDate = new \DateTime('-6 months');
                 $constraints[] = new Constraints\Range([
-                    'min' => new \DateTime('-6 months'),
+                    'min' => $minDate->modify('-1 day')->setTime(0, 0, 0),
                     'max' => new \DateTime('today'),
-                    'minMessage' => 'Date cannot be greater than six months in the past',
-                    'maxMessage' => 'Date cannot be in the future'
```

Comment:
The min and max message are valid options for the range constraint. It's not clear to me why they aren't used instead of the notInRangeMessage, but can we keep them here? If it's potentially a bug in the validator component that is later fixed, we want our custom messages to still be displayed.